lib/util/index.js

```diff
@@ -553,6 +553,59 @@ function valueToJSON(value) {
   return value;
 }
 
+function createRoomConnectEventPayload(connectOptions) {
+  function boolToString(val) {
+    return val ? 'true' : 'false';
+  }
+  const payload = {
+    // arrays props converted to lengths.
+    iceServers: (connectOptions.iceServers || []).length,
+    audioTracks: (connectOptions.tracks || []).filter(track => track.kind === 'audio').length,
+    videoTracks: (connectOptions.tracks || []).filter(track => track.kind === 'video').length,
+    dataTracks: (connectOptions.tracks || []).filter(track => track.kind === 'data').length,
+  };
+
+  // boolean properties.
+  [['audio'], ['automaticSubscription'], ['enableDscp'], ['eventListener'], ['preflight'], ['video'], ['dominantSpeaker', 'enableDominantSpeaker']].forEach(([prop, eventProp]) => {
+    eventProp = eventProp || prop;
+    payload[eventProp] = boolToString(!!connectOptions[prop]);
+  });
+
+  // numbers and string properties.
+  [['maxVideoBitrate'], ['maxAudioBitrate'], ['iceTransportPolicy'], ['region'], ['name', 'roomName']].forEach(([prop, eventProp]) => {
+    eventProp = eventProp || prop;
+    if (typeof connectOptions[prop] === 'number' || typeof connectOptions[prop] === 'string') {
+      payload[eventProp] = connectOptions[prop];
+    }
+  });
+
+  // array props stringified.
+  ['preferredAudioCodecs', 'preferredVideoCodecs'].forEach(prop => {
+    if (typeof connectOptions[prop] !== 'undefined') {
+      payload[prop] = JSON.stringify(connectOptions[prop]);
+    }
+  });
+
+  if (connectOptions.bandwidthProfile) {
+    payload.bandwidthProfileOptions = {};
+    ['mode', 'maxTracks', 'trackSwitchOffMode', 'dominantSpeakerPriority'].forEach(prop => {
+      if (typeof connectOptions.bandwidthProfile[prop] === 'number' || typeof connectOptions.bandwidthProfile[prop] === 'string') {
```

Comment:
@makarandp0 At this point we would have already validated the bandwidth profile options object. So you just need to check the presence of the property.
```suggestion
      if (prop in connectOptions.bandwidthProfile) {
```